Homework11/WebTableTests.cs

```diff
@@ -0,0 +1,226 @@
+            Assert.AreEqual(ExpectedConstants.NewEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        // Test4: Edit first entry.
+        [Test]
+        public void EditFirstEntry()
+        {
+            var editButton = _driver.FindElement(By.XPath("//span[@title='Edit']"));
+            editButton.Click();
+
+            // Find form inputs.
+            var fistNameInput = _driver.FindElement(By.XPath("//input[@id='firstName']"));
+            var lastNameInput = _driver.FindElement(By.XPath("//input[@id='lastName']"));
+            var emailInput = _driver.FindElement(By.XPath("//input[@id='userEmail']"));
+            var ageInput = _driver.FindElement(By.XPath("//input[@id='age']"));
+            var salaryInput = _driver.FindElement(By.XPath("//input[@id='salary']"));
+            var departmentInput = _driver.FindElement(By.XPath("//input[@id='department']"));
+            var submitButton = _driver.FindElement(By.XPath("//button[@id='submit']"));
+
+            // Clear form inputs.
+            DeleteValue(fistNameInput);
+            DeleteValue(lastNameInput);
+            DeleteValue(emailInput);
+            DeleteValue(ageInput);
+            DeleteValue(salaryInput);
+            DeleteValue(departmentInput);
+
+            // Fill in form inputs.
+            FillInInput(fistNameInput, Constants.FirstName);
+            FillInInput(lastNameInput, Constants.LastName);
+            FillInInput(emailInput, Constants.Email);
+            FillInInput(ageInput, Constants.Age);
+            FillInInput(salaryInput, Constants.Salary);
+            FillInInput(departmentInput, Constants.Department);
+            submitButton.Click();
+
+            // Get text of updated entry.
+            ReadOnlyCollection<IWebElement> gridcell = _driver.FindElements(By.XPath("((//div[@role='row'])[2])/div"));
+            Assert.AreEqual(ExpectedConstants.NewEntryValue, GetTextOfCollectionElements(gridcell));
+        }
+
+        // Test5: Delete first entry.
+        [Test]
+        public void DeleteFirstEntry()
+        {
+            // Get the first name from the first row.
+            var firstName1 = _driver.FindElement(By.XPath("((//div[@role='rowgroup'])[1]//descendant::div[@role='gridcell'])[1]")).Text;
+
+            // Remove the first row.
+            var deleteButton = _driver.FindElement(By.XPath("//span[@id='delete-record-1']"));
+            deleteButton.Click();
+
+            // Get the rows left after the delete operation.
+            ReadOnlyCollection<IWebElement> rows = _driver.FindElements(By.XPath("//div[@role='rowgroup']"));
+
+            // Get the rows from the collection above that contains the first name from removed entry.
+            var rowsWithRemovedValue = GetListOfFilteredRows(GetListOfNotEmptyRows(rows), firstName1);
+            Assert.AreEqual(0, rowsWithRemovedValue.Count);
```

Comment:
Not required for fixes... In some edge case it can fail - if there are more 1 row with the same text as in the first row

Reply:
yep, already thought about it
hope, I will not forget to take this into account in homework 13